### PR TITLE
Enable more advanced travis testing

### PR DESCRIPTION
**Pull request by bgruening**
This is a request for comments. As you may probably know I'm working on porting dependencies to [conda](https://github.com/galaxyproject/tools-iuc/issues/522) with the final aim of getting all tools under test by simply install the dependencies via conda before running `planemo test`.

Currently all tools listed in the [tool test blacklist](https://github.com/galaxyproject/tools-iuc/blob/travis/.tt_blacklist) are not tested with `planemo test`. A few tools like bedtools I already removed from the blacklist and they are under extensive testing if we merge this PR. The plan would be to get as many dependencies into conda as possible and shorten the blacklist over time.

Ideally, this should serve as good example of how Galaxy tools should be tested and we should work also on devteam-tools and similar repositories. Maybe a dedicated hackathon would be nice or a GSOC project.

Also speaking about dependencies we are currently working on [mulled](https://github.com/mulled/mulled). This will automatically transform Conda packages into Docker containers. The aim are PR like this one https://github.com/bgruening/galaxytools/commit/7a5239910fda9ed90cca286a38855703b40b1b56 which in combination with this [Dockerfile](https://github.com/bgruening/docker-recipes/blob/master/galaxy-openms/Dockerfile) will resolve dependencies via Docker.

This Image runs Docker as a dependency resolver inside Galaxy Docker and it's working!
```
docker run --rm -p 8080:80 -i -t --privileged=true quay.io/bgruening/galaxy-openms
```

So getting packages in conda does also mean we can automatically build Docker containers and offer Docker and Conda dependency resolutions for everyone **today** and for testing.

@peterjc I would love to have your opinion on this as you have much experience with testing Galaxy tools.

There are still some strange things we need to figure out, like 
 * this PR is failing for unknown reasons https://github.com/galaxyproject/tools-iuc/pull/567
 * Is there anyway we can display planemos html output in travis logs?
 * we need a newer planemo release
 * samtools and bcftools should be automatically resolved, currently they are needed in the base environment (ping @jmchilton )

### CODE REVIEW

**Comment by martenson** [comment]
yay !

**Comment by erasche** [comment]
@bgruening this is awesome work, I especially like that you're only testing changed repositories. :D

**Comment by martenson** [comment]
if we introduce blacklist that means that every new tool has to be added to it unless it is conda-ready (tm)?

**Comment by erasche** [comment]
Yep, just until we get to the point where `wc -l blacklist == 0` and we
can switch to requiring new tools to be conda ready

**Comment by nsoranzo** [comment]
@erasche I don't think we have to wait until the blacklist is empty, but surely some time is needed.

File: filter_directories.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
+#!/usr/bin/env python
+
+import os
+import sys
+
+bl = [tool.strip() for tool in open('.tt_blacklist', 'r')]
```
Comment (nsoranzo) [inline filter_directories.py]: https://stackoverflow.com/questions/18840880/will-using-list-comprehension-to-read-a-file-automagically-call-close

#nitpickingoftheday

File: .travis.yml [hunk 2]
```diff
@@ -1,9 +1,25 @@
 language: python
 python: 2.7
+
+before_install:
+  - export GALAXY_REPO=https://github.com/galaxyproject/galaxy
+  - export GALAXY_RELEASE=release_16.01
+  - export GALAXY_HOME=$HOME/galaxy
```
Comment (nsoranzo) [inline .travis.yml]: Is GALAXY_HOME used somewhere?

**Comment by bgruening** [comment]
@martenson correct, either adding to the blacklist or make them available in conda.
@erasche @nsoranzo I would suggest that at a given time (after 16.01 release?) every new tool should have a conda dep as well as a tool_dependencies ... after GCC2016 we can skip the tool_dependencies requirement and can go conda only it we want.

**Comment by bgruening** [comment]
Feel free to merge this if you agree that this is the way forward.

**Comment by nsoranzo** [comment]
:+1:

**Comment by mvdbeek** [comment]
I like it :+1:. 
Maybe it would be nice (for the future) to move both test and setup from .travis.yml into a script (and call it from tox?)

**Comment by bgruening** [comment]
@nsoranzo, @mvdbeek please feel free to merge it next time. I feel bad merging my own PR if they have a big impact.
@mvdbeek I don't mind to use tox, just wanted to keep simple at the beginning.

**Pull request merged**